src/box.jl

```diff
@@ -18,6 +18,9 @@ Box(c::SVector{N}, d::SVector{N},
 Box(c::AbstractVector, d::AbstractVector, axes=eye(length(c)), data=nothing) =
     (N = length(c); Box(SVector{N}(c), SVector{N}(d), SMatrix{N,N}(axes), data))
 
+Box(b::NTuple{2,AbstractVector}, axes=eye(length(b[1])), data=nothing) =
```

Comment:
I just copied the prototype, sorry if the variable names were confusing.  I mean `boxcorners(b1::AbstractVector, b2::AbstractVector, ...)`

The point is, you can define a constructor function that is not called `Box`.  (e.g. the `diagm` function is effectively a `Matrix` constructor).   It is probably clearer to use a new function name rather than relying on subtle distinctions in parenthesization to distinguish the different constructor arguments.